util/containerutil/shell_shared.go

```diff
@@ -291,12 +297,12 @@ func (sf *shellFrontend) commandContextOutput(ctx context.Context, args ...strin
 	return output, nil
 }
 
-func (sf *shellFrontend) supportsPlatformArg(ctx context.Context) bool {
-	// We can't run scratch, but the error is different depending on whether
-	// --platform is supported or not. This is faster than attempting to run
-	// an actual image which may require downloading.
-	output, _ := sf.commandContextOutput(ctx, "run", "--rm", "--platform", getPlatform(), "scratch")
-	return strings.Contains(output.string(), "Unable to find image")
+func (sf *shellFrontend) supportsPlatform(ctx context.Context, platform string) (bool, error) {
+	frontendInfo, err := sf.FrontendInformation(ctx)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to get platform information")
+	}
+	return frontendInfo.ServerPlatform == platform, nil
 }
```

Comment:
This function assumed if the error message `Unable to find image` was output it meant the platform was supported. That does not seem to be the case; At least, not anymore.